### PR TITLE
docs: update docs for release 0.4.2

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### CODE REVIEW

File: docs/quick_start/README.md [hunk 1]
```diff
@@ -22,12 +22,12 @@ docker run -d --name supernode --restart=always -p 8001:8001 -p 8002:8002 dragon
 
 ```bash
 SUPERNODE_IP=`docker inspect supernode -f '{{.NetworkSettings.Networks.bridge.IPAddress}}'`
-docker run -d --name dfclient -p 65001:65001 dragonflyoss/dfclient:0.4.1 --registry https://index.docker.io --node $SUPERNODE_IP
+docker run -d --name dfclient --restart=always -p 65001:65001 -v /root/.small-dragonfly:/root/.small-dragonfly dragonflyoss/dfclient:0.4.2 --registry https://index.docker.io --node $SUPERNODE_IP
```
Comment (yeya24) [inline docs/quick_start/README.md]: Is it better to use `$HOME/.small-dragonfly:/root/.small-dragonfly` here?

File: docs/user_guide/multi_machines_deployment.md [hunk 2]
```diff
@@ -53,7 +52,8 @@ EOD
 ```bash
 docker run -d --name dfclient --restart=always -p 65001:65001 \
     -v /etc/dragonfly:/etc/dragonfly \
-    dragonflyoss/dfclient:0.4.0 --registry https://index.docker.io
+    -v /root/.small-dragonfly:/root/.small-dragonfly \
```
Comment (yeya24) [inline docs/user_guide/multi_machines_deployment.md]: Same with the previous one.

File: docs/user_guide/install_server.md [hunk 3]
```diff
@@ -25,12 +25,23 @@ When deploying with physical machines, the following conditions must be met.
...
 ---|---
 Git|1.9.1+
-JDK|1.7+
-Maven|3.0.3+
+Golang|1.12.x
 Nginx|0.8+
 
 ## Procedure - When Deploying with Docker
 
+### Get Supernode image
```
Comment (yeya24) [inline docs/user_guide/install_server.md]: I am wondering what is the correct spell. `Supernode` or `SuperNode` or `supernode`? `Dfdaemon` or `DFdaemon` or `dfdaemon`?

**Comment by yeya24** [comment]
LGTM

**Comment by Starnop** [comment]
> The files in `docs/cli_reference/` are generated automatically, it's unnecessary to change them in this pr.

Done. PTAL again. THX>

**Pull request merged**